SerialPrograms/Source/PokemonHome/Programs/PokemonHome_BoxSorting.cpp

```diff
@@ -642,12 +665,51 @@ void BoxSorting::program(SingleSwitchProgramEnvironment& env, SwitchControllerCo
                         }
                         boxes_data[get_index(box_nb, row, column)]->ot_id = ot_id;
 
-                        // NOTE edit when adding new struct members (detections go here likely)
-
                         // level_box
-                        // ot_box
+                        int level = OCR::read_number_waterfill(env.console, extract_box_reference(screen, level_box), 0xff000000, 0xff7f7f7f);
+                        if (level < 0 || level > 100) {
+                            dump_image(env.console, ProgramInfo(), "ReadSummary_Level", screen);
+                        }
+                        boxes_data[get_index(box_nb, row, column)]->level = (uint16_t)level;
+                        env.console.log("Level: " + std::to_string(level), COLOR_GREEN);
+
+                        // language
+                        std::string language = OCR::ocr_read(Language::English, extract_box_reference(screen, language_box));
+                        if (language.empty()) {
+                            dump_image(env.console, ProgramInfo(), "ReadSummary_Language", screen);
+                            env.console.log("Failed to read language from the box.", COLOR_RED);
+                        } else {
+                            language.erase(language.find_last_not_of(" \n\r\t") + 1);
+                            boxes_data[get_index(box_nb, row, column)]->language = language;
+                            env.console.log("Detected Language: " + language, COLOR_GREEN);
+                        }
                         // nature_box
+                        std::string nature = OCR::ocr_read(Language::English, extract_box_reference(screen, nature_box));

```

Comment:
for the nature, you should be giving the language used by pokemon home as natures from asiatic languages for example won't be readble using an english OCR (also, it'd be better to use a dictionnary)